Packs/GigamonThreatINSIGHT/Integrations/GigamonThreatINSIGHT/GigamonThreatINSIGHT.py

```diff
@@ -400,6 +401,80 @@ def formatEvents(r_json):
     return newData
 
 
+def getFirstFetch(firstFetch) -> dict[str, Any]:
+
+    pieces = firstFetch.strip().split()
+    unit = pieces[len(pieces) - 1]
+
+    try:
+        delta = arg_to_number(
+            arg=pieces[0],
+            arg_name='first_fetch',
+            required=False
+        )
+    except Exception:
+        delta = 0
+
+    days: int = delta if delta and unit == 'days' else 0
+    hours: int = delta if delta and unit == 'hours' else 0
+
+    if days == 0 and hours == 0:
+        days = 7
+
+    return {'days': days, 'hours': hours}
+
+
+def mapSeverity(severity) -> int:
+    match severity:
+        case 'high':
+            return 3
+        case 'moderate':
+            return 2
+        case 'low':
+            return 1
+        case _:
+            return 0
+
+
+def getIncidents(result, last_fetch) -> Tuple[Dict[str, int], List[dict[str, Any]]]:
+    # Initialize an empty list of incidents to return
+    # Each incident is a dict with a string as a key
+    incidents: List[Dict[str, Any]] = []
+
+    last_incident_time = last_fetch
+    for detection in result.outputs:
+        incident_time = datetime.strptime(detection['first_seen'], DATE_FORMAT)
+
+        # Check if inciden has been reported before
+        if last_fetch >= incident_time:
+            continue
+
+        severity = mapSeverity(detection['rule_severity'])
+
+        incident = {
+            'name': detection['rule_name'],
+            'occurred': detection['first_seen'],
```

Comment:
Would "created" be better than "first_seen"? The detection should have been seen many times, but this particular one could be later in time.

Reply:
I am not sure about this since it was implemented this way before.